pkg/mcp/server/server.go

```diff
@@ -198,9 +202,19 @@ func (s *Server) StreamAggregatedResources(stream mcp.AggregatedMeshConfigServic
 	defer con.close()
 	go con.receive()
 
+	// fan-in per-type response channels into single response channel for the select loop below.
+	responseC := make(chan *WatchResponse)
+	for _, ch := range con.watches {
+		go func(in chan *WatchResponse) {
+			for v := range in {
+				responseC <- v
+			}
+		}(ch.responseC)
```

Comment:
can this solve the issue completely ?

Reply:
AFAICT, the root cause of the deadlock was sharing a single channel across watches (per type_url).   The request processing path for one type could become blocked if another type's response push was in flight.